pkg/chains/objects/objects.go

```diff
@@ -0,0 +1,146 @@
+package objects
+
+import (
+	"context"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const PipelineTaskLabel = "tekton.dev/pipelineTask"
+
+// Used as a generic Kubernetes object
+// Holds fields common to all objects
+// ref: https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.4/pkg/client#Object
+type Object interface {
+	metav1.Object
+	runtime.Object
+}
+
+type Result struct {
+	Name  string
+	Type  v1beta1.ResultsType
+	Value v1beta1.ArrayOrString
+}
+
+// Extending the generic Kubernetes object
+// Functions apply to all Tekton objects
+type TektonObject interface {
+	Object
+	GetKind() string
+	GetObject() interface{}
+	GetLatestAnnotations(ctx context.Context, clientSet versioned.Interface) (map[string]string, error)
+	Patch(ctx context.Context, clientSet versioned.Interface, patchBytes []byte) error
+	GetResults() []Result
+	GetServiceAccountName() string
+}
```

Comment:
Fine with keeping this here, but do you think any of this would be generally useful to contribute to the upstream pipelines library? 👀

Reply:
Yes, I think that's a good idea. Do you think it's okay to be apart of this PR for now, then separate it out later? Let me know your thoughts.